pkg/engine/operation/resource_node.go

```diff
@@ -23,59 +23,67 @@ var _ ExecutableNode = (*ResourceNode)(nil)
 
 func (rn *ResourceNode) Execute(operation *Operation) status.Status {
 	log.Debugf("execute node:%s", rn.ID)
-	// 1. prepare planedState
-	planedState := rn.state
-	if rn.Action != Delete {
-		if operation.OperationType != Preview {
-			// replace implicit references
-			value := reflect.ValueOf(rn.state.Attributes)
-			_, implicitValue, s := ParseImplicitRef(value, operation.CtxResourceIndex, ImplicitReplaceFun)
-			if status.IsErr(s) {
-				return s
-			}
-			rn.state.Attributes = implicitValue.Interface().(map[string]interface{})
+	if operation.OperationType == Apply {
+		// replace implicit references
+		value := reflect.ValueOf(rn.state.Attributes)
+		_, implicitValue, s := ParseImplicitRef(value, operation.CtxResourceIndex, ImplicitReplaceFun)
+		if status.IsErr(s) {
+			return s
 		}
-	} else {
-		planedState = nil
+		rn.state.Attributes = implicitValue.Interface().(map[string]interface{})
 	}
 
+	// 1. prepare planedState
+	planedState := rn.state
+
 	// 2. get prior state which is stored in kusion_state.json
 	key := rn.state.ResourceKey()
 	priorState := operation.PriorStateResourceIndex[key]
 
-	// get the latest resource from runtime
-	liveState, s := operation.Runtime.Read(context.Background(), priorState)
+	// 3. get the latest resource from runtime
+	liveState, s := operation.Runtime.Read(context.Background(), planedState)
 	if status.IsErr(s) {
 		return s
 	}
 
-	// 3. compute ActionType of current resource node between planState and liveState
-	if liveState == nil {
-		rn.Action = Create
-	} else if planedState == nil {
+	// 4. compute ActionType of current resource node between planState and liveState
+	switch operation.OperationType {
```

Comment:
Isn't it enough with comparing betweent states to determinate the resource's action? Why we should import OperationType in resource_node?

Reply:
I have these thoughts:
1、we have two operations which will make changes on resource, one is apply, other is destroy
2、destroy operation doesn't need compare, it must be Delete action 
3、apply need compare, cause kusion apply contains create/update/delete action